### ISSUE TITLE
Support arm64 (M1) architecture in provided image

### ISSUE THREAD

**Issue by philipkozeny** [mediagis/nominatim-docker#198]
Great work guys! I'm working from a Macbook with M1. Can you check M1 support? I'm getting the following warning:
`WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested`

_Originally posted by @bruno-hl in https://github.com/mediagis/nominatim-docker/issues/197#issuecomment-817955718_

**Comment by bruno-hl** [comment]
Thanks @philipkozeny for opening this for me. I can help with some testing if you need!

**Comment by philipkozeny** [comment]
@bruno-hl for now you could try to build the image by yourself on your device and report back if that works (https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md)

@leonardehrenfried seems like this would be the way to go with our actions: https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md

Although I'm not 100% sure since we are not using the build & push rule. Otherwise we could use `buildx` instead of `build` directly.

**Comment by leonardehrenfried** [comment]
@bruno-hl If you want to help we would love a pull request to build this using Github Actions.

**Comment by leonardehrenfried** [comment]
@philipkozeny I have no clue how crossbuilding works and how fast it is but if it's not too much of a slowdown it would be but after reading the docs, I would do the following:

1. Setup qemu
2. Setup buildx
3. Build using buildx with the required platforms
4. Continue using the current CI pipeline

Do you have to do anything special to push a multiplatform image or will that just work?

**Comment by philipkozeny** [comment]
Sounds like a plan, although I have to admit that I also have no clue :D 

As far as I have seen there is nothing special to consider when pushing a multiplatform image, guess we just need to try it out. I'll try to get hold of a M1 device to test if everything works.

**Comment by mtmail** [comment]
https://www.scaleway.com/en/hello-m1/ 0.10 Euro/hour, 24h minimum.

**Comment by bruno-hl** [comment]
Thanks guys. I have no time this week but I can try to help next if still needed. I'll try to make some time.

**Comment by philipkozeny** [comment]
FYI: I've started to work on a PR, will provide an image to test soon.

**Comment by philipkozeny** [comment]
Can someone with an M1 mac (@bruno-uy ?) please use the command to test a multiplatform image I've provided:

```
docker run -it --rm \
  -e PBF_URL=https://download.geofabrik.de/europe/monaco-latest.osm.pbf \
  -e REPLICATION_URL=https://download.geofabrik.de/europe/monaco-updates/ \
  -p 8080:8080 \
  --name nominatim \
  phluko/nominatim:3.7
```

Should work in theory now. Don't use my image in production, I will delete it after we merge the PR and provide the official image at mediagis.

**Comment by bruno-uy** [comment]
Sorry for the delay @philipkozeny . Let me test it now!

**Comment by philipkozeny** [comment]
> > Can someone with an M1 mac (@bruno-uy ?) please use the command to test a multiplatform image I've provided:
> > ```
> > docker run -it --rm \
> >   -e PBF_URL=https://download.geofabrik.de/europe/monaco-latest.osm.pbf \
> >   -e REPLICATION_URL=https://download.geofabrik.de/europe/monaco-updates/ \
> >   -p 8080:8080 \
> >   --name nominatim \
> >   phluko/nominatim:3.7
> > ```
> > 
> > 
> > Should work in theory now. Don't use my image in production, I will delete it after we merge the PR and provide the official image at mediagis.
> 
> Tested and working properly with http://localhost:8080/search.php?q=avenue%20pasteur. Result:
> 
> ```json
> [{"place_id":102053,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright","osm_type":"way","osm_id":158189814,"boundingbox":["43.7307864","43.7308489","7.4149157","7.4149444"],"lat":"43.730819","lon":"7.4149444","display_name":"Avenue Pasteur, Fontvieille, Mónaco, 98020, Mónaco","place_rank":26,"category":"highway","type":"residential","importance":0.3},{"place_id":100741,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright","osm_type":"way","osm_id":92627433,"boundingbox":["43.7308657","43.7311052","7.4131392","7.4146148"],"lat":"43.7308657","lon":"7.4131392","display_name":"Avenue Pasteur, Jardin Exotique, Mónaco, 98020, Mónaco","place_rank":26,"category":"highway","type":"residential","importance":0.3}]
> ```

Thanks for testing the image, it's now available at ```mediagis/nominatim:3.6``` & ```mediagis/nominatim:3.7```

**Comment by leonardehrenfried** [comment]
Wow, cross-building takes an hour?
